skimage/io/collection.py

```diff
@@ -72,20 +78,31 @@ def alphanumeric_key(s):
     return k
 
 
-class ImageCollection(object):
+def _is_multipattern(input_pattern):
+    """Helping function. Returns True if pattern contains a tuple, list, or a
+    string separated with os.pathsep."""
+    # Conditions to be accepted by ImageCollection:
+    has_str_ospathsep = (isinstance(input_pattern, str)
+                         and os.pathsep in input_pattern)
+    not_a_string = not isinstance(input_pattern, str)
+    has_iterable = isinstance(input_pattern, Sequence)
+    has_strings = all(isinstance(pat, str) for pat in input_pattern)
 
-    """Load and manage a collection of image files.
+    is_multipattern = (has_str_ospathsep or
+                       (not_a_string
+                        and has_iterable
+                        and has_strings))
+    return is_multipattern
 
-    Note that files are always stored in alphabetical order. Also note that
-    slicing returns a new ImageCollection, *not* a view into the data.
+
+class ImageCollection(object):
+    """Load and manage a collection of image files.
 
     Parameters
     ----------
     load_pattern : str or list
```

Comment:
This description is not complete.

Reply:
I thought a little and I think you're right. Let's maintain the old behavior but without the documentation, to not break stuff for a while but not encouraging it.